src/core/Akka.Persistence.TCK/Snapshot/SnapshotStoreSpec.cs

```diff
@@ -198,6 +201,54 @@ public virtual void SnapshotStore_should_delete_a_single_snapshot_identified_by_
                 && result.Snapshot.Snapshot.ToString() == "s-2");
         }
 
+        // Issue #7312
+        // Backward compatibility check, old SnapshotMetadata .ctor default value should work as expected
+        [Fact]
+        public virtual void SnapshotStore_should_delete_a_single_snapshot_identified_by_SequenceNr_in_snapshot_metadata_if_timestamp_is_MinValue()
+        {
+            var md = Metadata[2];
+            // In previous incarnation, timestamp argument defaults to DateTime.MinValue
+            md = new SnapshotMetadata(md.PersistenceId, md.SequenceNr, DateTime.MinValue);
+            var command = new DeleteSnapshot(md);
+            var sub = CreateTestProbe();
+
+            Subscribe<DeleteSnapshot>(sub.Ref);
+            SnapshotStore.Tell(command, _senderProbe.Ref);
+            sub.ExpectMsg(command);
+            _senderProbe.ExpectMsg<DeleteSnapshotSuccess>();
+
+            SnapshotStore.Tell(new LoadSnapshot(Pid, new SnapshotSelectionCriteria(md.SequenceNr), long.MaxValue), _senderProbe.Ref);
+            _senderProbe.ExpectMsg<LoadSnapshotResult>(result =>
+                result.ToSequenceNr == long.MaxValue
+                && result.Snapshot != null
+                && result.Snapshot.Metadata.Equals(Metadata[1])
+                && result.Snapshot.Snapshot.ToString() == "s-2");
+        }
+        
+        // Issue #7312, this is a side effect of the ctor signature changes
+        // DeleteSnapshot should not delete snapshot if timestamp value does not meet deletion criteria
+        [Fact]
+        public virtual void SnapshotStore_should_not_delete_snapshot_identified_by_SequenceNr_if_metadata_timestamp_is_less_than_stored_timestamp()
```

Comment:
This is a side effect of the changes made in this PR.
User need to either have a `SnapshotMetadata.Timestamp` that is greater than the stored value, or equals to `DateTime.MinValue` for the actual `Snapshot` to be deleted.

Reply:
We either have this as an actual spec, or change the code to disregard `SnapshotMetadata.Timestamp` altogether when deleting a `Snapshot`.

Reply:
Should we add an overrideable `virtual` variable that allows this spec to be enable / disable on an implementation by implementation basis?

Reply:
The unit test methods themselves are virtual, and this is part of the spec now.
I think it is reasonable to force the other SQL implementation to adhere to the new spec?

Reply:
Well, "force", someone can always override and skip the test in implementation

Reply:
Doh, I missed that. You're right. LGTM.